sql/testdata/select_non_covering_index

```diff
@@ -63,3 +63,25 @@ EXPLAIN SELECT * FROM t WHERE c > 0 ORDER BY c
 0 index-join
 1 scan       t@c /1-
 1 scan       t@primary
+
+statement ok
+INSERT INTO t VALUES (7, 8, NULL)
+
+query I
+SELECT c FROM t WHERE a=7
+----
+NULL
```

Comment:
Seems like there is another bug somewhere. Why is selecting a `NULL` value causing a crash?  This also seems unrelated to non-covering indexes as you're using the primary index here.